src/scenarios/billing/SubscribeToSubscriptionPlan/SubscribeToSubscriptionPlan.tsx

```diff
@@ -0,0 +1,476 @@
+                    }
+                  }
+                },
+              })
+              .render('#paypal-btn-container');
+          }
+        } catch (error) {
+          if (error instanceof Error) {
+            logger.error(
+              'calling [subscribeToPaymentPlan]',
+              'output:',
+              'failed to load the PayPal JS SDK script. ' + error.message,
+            );
+          }
+        }
+        setPaypalInitializing(false);
+      } else {
+        logger.error(
+          'Please provide values to all the input fields above to use popup based payment flow.',
+        );
+      }
+    } else {
+      setPurchaseFlow(newPurchaseFlow);
+    }
+  };
+
+  const subscribeToPaymentPlan = async (): Promise<void> => {
+    if (purchaseFlow === 'REDIRECT') {
+      try {
+        const result = await apolloClient.query({
+          query: subscribeRedirectMutation,
+          variables: {
+            input: {
+              paymentPlanId,
+              purchaseFlow: 'REDIRECT',
+              country,
+            },
+          },
+          context: {
+            headers: {
+              'Content-Type': 'application/json',
+              Authorization: `Bearer ${userAccessToken}`,
+            },
+          },
+          fetchPolicy: 'no-cache',
+        });
+
+        logger.log('calling [subscribeToPaymentPlan]', 'output:', result.data);
+        if (result.errors) {
+          logger.error(
+            'calling [subscribeToPaymentPlan]',
+            'output:',
+            result.errors,
+          );
+        } else {
+          window.location.href = result.data.paypalSubscribe.approveUrl;
+        }
+      } catch (error) {
+        setPaymentPlanPrices([]);
+        if (error instanceof Error) {
+          logger.error(
+            'calling [subscribeToPaymentPlan]',
+            'output:',
+            error.message,
+          );
+        }
+      }
+    }
+  };
+
+  return (
+    <>
+      <Segment basic>
+        <Header size="huge">
+          Subscribe to a Subscription Plan (with Paypal purchase)
+        </Header>
+        <Header size="small">
+          Required Services:
+          <Label>user-service</Label>
+          <Label>billing-service</Label>
+        </Header>
+
+        <Divider />
+
+        <Container fluid>
+          <p>
+            This scenario demonstrates the purchase flow of a PayPal-based
+            payment plan for a logged-in user.
+          </p>
+          <p>
+            If the user is not already signed-in, you can use one of the Sign-In
+            scenarios.
+          </p>
+          <p>
+            Billing Service should be configured with the following settings for
+            this scenario to work:
+          </p>
+          <p>
+            <b>Success Redirect URL:</b> {window.location.href}
+            /subscribe-success
+          </p>
+          <p>
+            <b>Error Redirect URL:</b> {window.location.href}/subscribe-error
+          </p>
```

Comment:
the `location.href` is susceptible for spoofed URLs here. we can consider a more methodically built URL approach here?

generating redirect URLs using `req.basePath` + `scenarioId` + `fixedConst`